lib/swiftype/request.rb

```diff
@@ -57,7 +57,14 @@ def request(method, path, params={})
         uri = URI.parse("#{Swiftype.endpoint}#{path}")
 
         request = build_request(method, uri, params)
-        http = Net::HTTP.new(uri.host, uri.port)
+
+        if proxy
+          proxy_parts = URI.parse(proxy)
+          http = Net::HTTP.new(uri.host, uri.port, proxy_parts.host, proxy_parts.port)
```

Comment:
Mind passing the username / password through as well?

```suggestion
          http = Net::HTTP.new(uri.host, uri.port, proxy_parts.host, proxy_parts.port, proxy_parts.user, proxy_parts.password)
```